src/v/cluster/partition_manager.cc

```diff
@@ -111,7 +111,8 @@ ss::future<consensus_ptr> partition_manager::manage(
   std::vector<model::broker> initial_nodes,
   std::optional<remote_topic_properties> rtp,
   std::optional<cloud_storage_clients::bucket_name> read_replica_bucket,
-  raft::with_learner_recovery_throttle enable_learner_recovery_throttle) {
+  raft::with_learner_recovery_throttle enable_learner_recovery_throttle,
+  raft::keep_snapshotted_log keep_snapshotted_log) {
```

Comment:
we probably need an escape hatch to skip snapshots and replay directly from log (not a blocker for this patch)?